src/main/java/de/vogel612/testclient_javabot/TestProgram.java

```diff
@@ -39,14 +39,23 @@
 
 	public TestProgram() {
 		LOGGER.info("instantiating TestProgram class");
-		AppContext.INSTANCE.add(chatInterface);
-		bot = new DefaultBot(chatInterface);
-		gui = new ClientGui();
-		chatInterface.subscribe(bot);
-		chatInterface.subscribe(gui);
-		LOGGER.info("Basic component setup completed, beginning command glueing.");
-		new CoreBotCommands(chatInterface, bot).allCommands().forEach(bot::subscribe);
-		LOGGER.info("Glued Core Commands");
+        try{
+	        AppContext.INSTANCE.add(chatInterface);
+	        bot = new DefaultBot(chatInterface);
+	        Executors.newSingleThreadExecutor().execute(() -> {
+	        	ClientGui.launch(ClientGui.class);
+			});
+	        ClientGui.latch.await();
+	        gui = ClientGui.getInstance();
+	        chatInterface.subscribe(bot);
+	        chatInterface.subscribe(gui);
+	        LOGGER.info("Basic component setup completed, beginning command glueing.");
+	        new CoreBotCommands(chatInterface, bot).allCommands().forEach(bot::subscribe);
+        }
+        catch(InterruptedException die) {
```

Comment:
the try block should be as small as possible. the only place where the InterruptedException can be thrown is `ClientGui.latch.await()`. if you `throw new RuntimeException(die);` anyways it doesn't execute the rest anyways ;)